### PR TITLE
Update `test_pg_headroom_update.py` to use new function from `gu_utils.py`

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

Previously, we use function `duthost.get_asic_name` in test case `test_pg_headroom_update` to get the asic name of testbed. In this function, somehow, it considers the asic name of kvm testbed 'unknown'. But in the next step, in function `is_valid_platform_and_version`, it uses the function `get_asic_name` from file `generic_config_updater/gu_utils.py`, and considers the asic name of kvm testbed 'td2'.The difference will cause the misjudgment of the result `expect_op_success` and `expect_op_failure`. To keep consistent, we will use function `get_asic_name` from  `generic_config_updater/gu_utils.py`.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
Previously, we use function `duthost.get_asic_name` in test case `test_pg_headroom_update` to get the asic name of testbed. In this function, somehow, it considers the asic name of kvm testbed 'unknown'. But in the next step, in function `is_valid_platform_and_version`, it uses the function `get_asic_name` from file `generic_config_updater/gu_utils.py`, and considers the asic name of kvm testbed 'td2'.The difference will cause the misjudgment of the result `expect_op_success` and `expect_op_failure`. To keep consistent, we will use function `get_asic_name` from  `generic_config_updater/gu_utils.py`.

#### How did you do it?
Use the function `get_asic_name` from `generic_config_updater/gu_utils.py` to instead the function `duthost.get_asic_name`

### CODE REVIEW

File: .azure-pipelines/pr_test_scripts.yaml [hunk 1]
```diff
@@ -333,6 +332,12 @@ onboarding_t0:
   - test_pktgen.py
   - arp/test_unknown_mac.py
   - hash/test_generic_hash.py
+  - dualtor/test_orchagent_active_tor_downstream.py
+  - dualtor/test_orchagent_mac_move.py
+  - dualtor/test_orchagent_standby_tor_downstream.py
+  - dualtor/test_standby_tor_upstream_mux_toggle.py
```
Comment (xwjiang-ms) [inline .azure-pipelines/pr_test_scripts.yaml:338]: need to sync with latest master, these dualtor scripts were put in to t0 job

Comment (yutongzhang-microsoft) [reply .azure-pipelines/pr_test_scripts.yaml:338]: > need to sync with latest master, these dualtor scripts were put in to t0 job

Sure

File: tests/generic_config_updater/test_pg_headroom_update.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def _confirm_value_in_app_and_asic_db():
 
 @pytest.mark.parametrize("operation", ["replace"])
 def test_pg_headroom_update(duthost, ensure_dut_readiness, operation, skip_when_buffer_is_dynamic_model):
-    asic_type = duthost.get_asic_name()
+    asic_type = get_asic_name(duthost)
```
Comment (opcoder0) [inline tests/generic_config_updater/test_pg_headroom_update.py:82]: `sonic-cfggen -d -v DEVICE_METADATA.localhost.hwsku` on vs seems to be recognized as `Force10-S6000` which leads to `td2`. May be we can add a comment?

**Review by opcoder0 — APPROVED** [review]
LGTM

**Review by wangxin — APPROVED** [review]

**Pull request merged**